src/Uno.Extensions.Reactive/Core/ListState.Extensions.cs

```diff
@@ -196,10 +210,36 @@ public static IDisposable Execute<T>(this IListState<T> state, AsyncAction<IImmu
 	/// <param name="action">The callback to invoke on each update of the state.</param>
 	/// <param name="caller"> For debug purposes, the name of this subscription. DO NOT provide anything here, let the compiler fulfill this.</param>
 	/// <param name="line">For debug purposes, the name of this subscription. DO NOT provide anything here, let the compiler fulfill this.</param>
-	/// <returns>A <see cref="IDisposable"/> that can be used to remove the callback registration.</returns>
-	public static IDisposable ForEachAsync<T>(this IListState<T> state, AsyncAction<IImmutableList<T>> action, [CallerMemberName] string? caller = null, [CallerLineNumber] int line = -1)
+	/// <returns>A <see cref="IListState{T}"/> that can be used to chain other operations.</returns>
+	public static IListState<T> ForEach<T>(this IListState<T> state, AsyncAction<IImmutableList<T>> action, [CallerMemberName] string? caller = null, [CallerLineNumber] int line = -1)
 		where T : notnull
-		=> new StateForEach<IImmutableList<T>>(state, (list, ct) => action(list.SomeOrDefault() ?? ImmutableList<T>.Empty, ct), $"ForEachAsync defined in {caller} at line {line}.");
+	{
+		_ = AttachedProperty.GetOrCreate(owner: state,
+										 key: action,
+										 factory: (s, ks) => new StateForEach<IImmutableList<T>>(state, (list, ct) => action(list.SomeOrDefault() ?? ImmutableList<T>.Empty, ct), $"ForEachAsync defined in {caller} at line {line}."));
```

Comment:
```suggestion
										 factory: static (s, ks) => new StateForEach<IImmutableList<T>>(state, (list, ct) => action(list.SomeOrDefault() ?? ImmutableList<T>.Empty, ct), $"ForEach defined in {caller} at line {line}."));
```

Same for each overloads + `State.Extensions.cs` ... (+ styling :P )